File: _objc/NetworkingObjC/NetworkingObjC/Presentation/Post/PostView.m
//
//  PostView.m
//  NetworkingObjC
//
//  Created by Bondar Yaroslav on 25/06/2017.
//  Copyright © 2017 Bondar Yaroslav. All rights reserved.
//

#import "PostView.h"

@interface PostView ()

@property (weak, nonatomic) IBOutlet UILabel *idLabel;
@property (weak, nonatomic) IBOutlet UILabel *userIdLabel;
@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
@property (weak, nonatomic) IBOutlet UILabel *bodyLabel;

@end

@implementation PostView

static NSString * const longTextForScroll = @"Objective-C is a general-purpose, object-oriented programming language that adds Smalltalk-style messaging to the C programming language. It was the main programming language used by Apple for the OS X and iOS operating systems, and their respective application programming interfaces (APIs) Cocoa and Cocoa Touch prior to the introduction of Swift. The programming language Objective-C was originally developed in the early 1980s. It was selected as the main language used by NeXT for its NeXTSTEP operating system, from which OS X and iOS are derived.[3] Portable Objective-C programs that do not use the Cocoa or Cocoa Touch libraries, or those using parts that may be ported or reimplemented for other systems, can also be compiled for any system supported by GNU Compiler Collection (GCC) or Clang.";

- (void)fillWith:(Post *)post {

    
    self.idLabel.text = [@(post.id) stringValue]; /// 1st way int to string
    self.userIdLabel.text = [NSString stringWithFormat:@"%ld", (long)post.userId]; /// 2nd way int to string
    self.titleLabel.text = post.title;
    self.bodyLabel.text =  [NSString stringWithFormat:@"%@\n\n\n%@", post.body, longTextForScroll];
}

@end

File: ViperApp/ViperApp/Presentations/UserStories/Detail/View/PostView.swift
//
//  PostView.swift
//  ViperApp
//
//  Created by Bondar Yaroslav on 30/09/2017.
//  Copyright © 2017 Bondar Yaroslav. All rights reserved.
//

import UIKit

final class PostView: UIView {
    @IBOutlet private weak var titleLabel: UILabel!
    @IBOutlet private weak var idLabel: UILabel!
    @IBOutlet private weak var userIdLabel: UILabel!
    @IBOutlet private weak var bodyLabel: UILabel!
    
    func fill(with post: Post) {
        titleLabel.text = post.title
        idLabel.text = "ID: \(post.id)"
        userIdLabel.text = "UserID: \(post.userId)"
        bodyLabel.text = post.body
    }
}
